lib/js-file.js

```diff
@@ -110,24 +123,25 @@ JsFile.prototype = {
      * Returns whether a specific rule is disabled on the given line.
      *
      * @param {String} ruleName the rule name being tested
-     * @param {Number} line the line number being tested
+     * @param {Element} element being tested
      * @returns {Boolean} true if the rule is enabled
      */
-    isEnabledRule: function(ruleName, line) {
-        var enabled = true;
-        this._disabledRuleIndex.some(function(region) {
-            // once the comment we're inspecting occurs after the location of the error,
-            // no longer check for whether the state is enabled or disable
-            if (region.line > line) {
-                return true;
-            }
-
-            if (region.rule === ruleName || region.rule === '*') {
-                enabled = region.enabled;
-            }
-        }, this);
-
-        return enabled;
+    isEnabledRule: function(ruleName, element) {
```

Comment:
Are you planning to rework these methods as well? Seems unrelated to proposed changes?